crates/fuel-core/src/query.rs

```diff
@@ -7,6 +8,21 @@ mod message;
 mod subscriptions;
 mod tx;
 
+pub trait QueryData:
+    Send
+    + Sync
+    + BalanceQueryData
+    + BlockQueryData
+    + ChainQueryData
+    + CoinQueryData
+    + ContractQueryData
+    + MessageQueryData
+    + MessageProofData
+    + TransactionQueryData
+{
+}
+impl<D: DatabasePort + 'static> QueryData for D {}
```

Comment:
```suggestion
impl<D> QueryData for D 
where D:
     Send
     + Sync
     + BalanceQueryData
     + BlockQueryData
     + ChainQueryData
     + CoinQueryData
     + ContractQueryData
     + MessageQueryData
     + MessageProofData
     + TransactionQueryData
     {}
```

I guess it should be something like this